robocop/checkers/naming.py

```diff
@@ -26,20 +27,20 @@ class InvalidCharactersInNameChecker(VisitorChecker):
 
     rules = {
         "0301": (
-            "invalid-char-in-name",
-            "Invalid character '%s' in %s name",
+            "not-allowed-char-in-name",
+            "Not allowed character '%s' found in %s name",
             RuleSeverity.WARNING,
             (
-                "invalid_chars",
-                "invalid_chars",
-                set,
-                "set of characters not allowed in a name",
+                "pattern",
+                "pattern",
+                pattern_type,
+                "pattern defining characters (not) allowed in a name",
             ),
         )
     }
 
     def __init__(self):
-        self.invalid_chars = {".", "?"}
+        self.pattern = re.compile(r"[\.\?]")
         super().__init__()
```

Comment:
We will need to extensively cover this change in release notes

Reply:
I already added it. Take a look :)